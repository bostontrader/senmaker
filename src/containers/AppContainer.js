import {Container} from 'flux/utils'
import {Map}       from 'immutable'

import AppActions     from '../data/AppActions'
import AppStore       from '../data/AppStore'

import NoundAEStore   from '../data/dictionary/nound/addedit/NoundAEStore'
import NoundAEActions from '../data/dictionary/nound/addedit/NoundAEActions'
import NoundStore     from '../data/dictionary/nound/NoundStore'

import VerbdAEStore   from '../data/dictionary/verbd/addedit/VerbdAEStore'
import VerbdAEActions from '../data/dictionary/verbd/addedit/VerbdAEActions'
import VerbdStore     from '../data/dictionary/verbd/VerbdStore'

import NouniActions   from '../data/nouni/NouniActions'
import NouniAEStore   from '../data/nouni/NouniAEStore'
import NouniStore     from '../data/nouni/NouniStore'

import StringActions  from '../data/StringActions'
import StringStore    from '../data/StringStore'

import AppView from '../views/AppView'

function getStores() {
    return [
        AppStore,
        NoundStore,
        NoundAEStore,
        NouniStore,
        NouniAEStore,
        StringStore,
        VerbdStore,
        VerbdAEStore
    ]
}

function getState() {

    // It's tempting to try to make this a single atom of immutable state. However, the immutable object
    // you try to create here will be converted by deus ex machina into an ordinary Object, devoid of the
    // immutable methods.
    return {
        lang: AppStore.getState().get('lang'),

        level: AppStore.getState().get('level'),
        onLevelPrevious: AppActions.levelPrevious,
        onLevelNext: AppActions.levelNext,
        onLevelReset: AppActions.levelReset,
        onSetQuizScore: AppActions.setQuizScore,

        // instantiated nound
        nouni: AppStore.getState().get('nouni'),
        onChangeDefiniteness: NouniActions.changeDefiniteness,
        //onChangeSelectedNoun: (n) => {console.log('AppContainer onSelectedNounChanged =',n)},
        //onChangeSelectedNoun: NouniActions.changeSelectedNoun, // move this to nound?
        //onInsertNouni: AppActions.insertNouni,
        addEditNouni: NouniAEStore.getState(),
        //onAddNouni: NouniDictionaryItemActions.addNouni,
        //onCancelNouni: NouniDictionaryItemActions.cancel,
        //onChangeNouniBase: NouniDictionaryItemActions.changeBase,
        //onDeleteNouni: NouniDictionaryItemActions.deleteNouni,
        //onClickEditNouni: NouniDictionaryItemActions.editNouni,
        //onInsertNouni: NouniDictionaryItemActions.insertNouni,
        //onUpdateNouni: NouniDictionaryItemActions.updateNouni,

        strings: StringStore.getState(),
        onLangEng: StringActions.langEng,
        onLangChn: StringActions.langChn,

        // noun dictionary
        nound: Map({
            addEditNound: NoundAEStore.getState(),
            nouns: NoundStore.getState(),
            //selectedNounId: NoundStore.getState().get('selectedNounId'),
            //onChangeSelectedNoun: NoundActions.changeSelectedNoun,

            onClickAddNound: NoundAEActions.clickAddNound,
            onClickCancel: NoundAEActions.clickCancel, // target only the NoundAEStore
            onClickDeleteNound: NoundAEActions.clickDeleteNoun,
            onClickEditNound: NoundAEActions.clickEditNound,
            onClickSaveNound: NoundAEActions.clickSaveNound,
            onChangeNoundBase: NoundAEActions.onChangeBase,
        }),
        
        // verb dictionary
        verbd: Map({
            addEditVerbd: VerbdAEStore.getState(),
            verbs: VerbdStore.getState(),
            //selectedVerbId: VerbdStore.getState().get('selectedVerbId'),
            //onChangeSelectedVerb: VerbdActions.changeSelectedVerb,

            onClickAddVerbd: VerbdAEActions.clickAddVerbd,
            onClickCancel: VerbdAEActions.clickCancel, // target only the VerbdAEStore
            onClickDeleteVerbd: VerbdAEActions.clickDeleteVerbd,
            onClickEditVerbd: VerbdAEActions.clickEditVerbd,
            onClickSaveVerbd: VerbdAEActions.clickSaveVerbd,
            onChangeVerbdBase: VerbdAEActions.onChangeBase,
        })
    }
}

export default Container.createFunctional(AppView, getStores, getState)
