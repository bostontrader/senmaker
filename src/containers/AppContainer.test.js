import React from 'react'
import rtRenderer from 'react-test-renderer'

import AppContainer from './AppContainer'

describe('AppContainer', () => {

    // What is there to even test here?  All the action takes place on lower levels.
    // Test that instead.
    it('\'s Alive!', () => {
        expect(rtRenderer.create(<AppContainer />).toJSON()).toMatchSnapshot()
    })

})