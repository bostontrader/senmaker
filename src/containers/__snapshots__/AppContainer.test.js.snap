exports[`AppContainer 's Alive! 1`] = `
<div
  className="container">
  <div
    className="row">
    <div>
      <div
        className="help"
        style={
          Object {
            "border": "1px solid black",
            "margin": "5px",
          }
        }>
        <p>
          这个程序将帮助您建立一个英语文章。
        </p>
        <p>
          为了做到这一点你必须首先学习如何构建英语的小块（句子和段落）。
        </p>
        <p>
          然后,您可以将英语的小块组装成更大的组件。
        </p>
        <p>
          通过以下课程您将知道如何学习英语。
        </p>
        <p>
          你目前是0级，为了继续下一个阶段需要通过测验。
        </p>
      </div>
      <div
        className="quiz"
        style={
          Object {
            "border": "1px solid black",
            "margin": "5px",
          }
        }>
        <h3>
          测试
        </h3>
        <table>
          <tbody>
            <tr>
              <td>
                <p>
                  <input
                    checked={false}
                    id="iunderstandCheck"
                    onChange={[Function]}
                    type="checkbox" />
                  我明白
                </p>
              </td>
              <td>
                
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="lesson-navigator"
        style={
          Object {
            "border": "1px solid black",
            "margin": "5px",
          }
        }>
        <p>
          级
           
          0
        </p>
        
        <button
          id="lesson-next"
          onClick={[Function]}>
          下一級
        </button>
        <button
          id="level-reset"
          onClick={[Function]}>
          返回
        </button>
        <img
          alt=""
          height="20"
          id="enFlag"
          onClick={[Function]}
          src="/img/us_flag.gif"
          width="28" />
        
      </div>
    </div>
  </div>
</div>
`;
